src/lib/RequestParser.ts

```diff
@@ -148,28 +152,75 @@ export class RequestParser {
                     kind: RequestType.SIGN_MESSAGE,
                     appName: signMessageRequest.appName,
                     signer: signMessageRequest.signer
-                            ? Nimiq.Address.fromUserFriendlyAddress(signMessageRequest.signer)
-                            : undefined,
+                        ? Nimiq.Address.fromUserFriendlyAddress(signMessageRequest.signer)
+                        : undefined,
                     message: signMessageRequest.message,
                 } as ParsedSignMessageRequest;
-            case RequestType.CASHLINK:
-                const cashlinkRequest = request as CashlinkRequest;
-                if (cashlinkRequest.cashlinkAddress && cashlinkRequest.senderAddress) {
-                    console.warn(
-                        'CashlinkRequest.cashlinkAddress and CashlinkRequest.senderAddress should not both be defined.',
-                    );
+            case RequestType.CREATE_CASHLINK:
+                const createCashlinkRequest = request as CreateCashlinkRequest;
+                const senderAddress = 'senderAddress' in createCashlinkRequest
+                    ? Nimiq.Address.fromUserFriendlyAddress(createCashlinkRequest.senderAddress)
+                    : undefined;
+                const senderBalance = 'senderBalance' in createCashlinkRequest
+                    ? createCashlinkRequest.senderBalance
+                    : undefined;
+                if (senderBalance !== undefined && !Nimiq.NumberUtils.isUint64(senderBalance)) {
+                    throw new Error('Invalid Cashlink senderBalance');
+                }
+
+                const value = createCashlinkRequest.value;
+                if (value !== undefined && (!Nimiq.NumberUtils.isUint64(value) || value === 0)) {
+                    throw new Error('Malformed Cashlink value');
+                }
+
+                let message = 'message' in createCashlinkRequest ? createCashlinkRequest.message : undefined;
+                if (message !== undefined) {
+                    if (typeof message !== 'string') {
+                        throw new Error('Cashlink message must be a string');
+                    }
+                    const { result: truncated, didTruncate } = Utf8Tools.truncateToUtf8ByteLength(message, 255);
+                    if (!('autoTruncateMessage' in createCashlinkRequest && createCashlinkRequest.autoTruncateMessage)
+                        && didTruncate) {
+                        throw new Error('Cashlink must be shorter than 256 bytes or autoTruncateMessage must be '
```

Comment:
The word `message` is missing in the error message: "Cashlink **message** must be shorter than 256 bytes..."